instrumentation/log4j/log4j-2.13.2/library/src/main/java/io/opentelemetry/instrumentation/log4j/v2_13_2/LogEventMapper.java

```diff
@@ -0,0 +1,102 @@
+import io.opentelemetry.sdk.logs.LogBuilder;
+import io.opentelemetry.sdk.logs.data.Severity;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.time.Instant;
+import org.apache.logging.log4j.message.Message;
+
+final class LogEventMapper {
+
+  // Visible for testing
+  static final AttributeKey<String> ATTR_THROWABLE_MESSAGE =
+      AttributeKey.stringKey("throwable.message");
+
+  /**
+   * Map the {@link LogEvent} data model onto the {@link LogBuilder}. Unmapped fields include:
+   *
+   * <ul>
+   *   <li>Fully qualified class name - {@link LogEvent#getLoggerFqcn()}
+   *   <li>Thread name - {@link LogEvent#getThreadName()}
+   *   <li>Thread id - {@link LogEvent#getThreadId()}
+   *   <li>Thread priority - {@link LogEvent#getThreadPriority()}
+   *   <li>Thread priority - {@link LogEvent#getThreadPriority()}
+   *   <li>Thrown details (stack trace, class name) - {@link LogEvent#getThrown()}
+   *   <li>Marker - {@link LogEvent#getMarker()}
+   *   <li>Nested diagnostic context - {@link LogEvent#getContextStack()}
+   *   <li>Mapped diagnostic context - {@link LogEvent#getContextData()}
+   * </ul>
+   */
+  static void mapLogEvent(LogBuilder builder, LogEvent logEvent) {
+    // TODO: map the LogEvent more completely when semantic conventions allow it
+    AttributesBuilder attributes = Attributes.builder();
+
+    // message
+    Message message = logEvent.getMessage();
+    if (message != null) {
+      builder.setBody(message.getFormattedMessage());
+    }
+
+    // time
+    Instant instant = logEvent.getInstant();
+    if (instant != null) {
+      builder.setEpoch(
+          TimeUnit.MILLISECONDS.toNanos(instant.getEpochMillisecond())
+              + instant.getNanoOfMillisecond(),
+          TimeUnit.NANOSECONDS);
+    }
+
+    // level
+    Level level = logEvent.getLevel();
+    if (level != null) {
+      builder.setSeverity(levelToSeverity(level));
+      builder.setSeverityText(logEvent.getLevel().name());
+    }
+
+    // throwable
+    Throwable throwable = logEvent.getThrown();
+    if (throwable != null) {
+      attributes.put(ATTR_THROWABLE_MESSAGE, throwable.getMessage());
+    }
+
+    // span context
+    builder.setContext(Context.current());
+
+    builder.setAttributes(attributes.build());
+  }
+
+  private static Severity levelToSeverity(Level level) {
+    switch (level.getStandardLevel()) {
+      case ALL:
+        return Severity.TRACE;
+      case TRACE:
+        return Severity.TRACE2;
+      case DEBUG:
+        return Severity.DEBUG;
+      case INFO:
+        return Severity.INFO;
+      case WARN:
+        return Severity.WARN;
+      case ERROR:
+        return Severity.ERROR;
+      case FATAL:
+        return Severity.FATAL;
+      case OFF:
+        return Severity.UNDEFINED_SEVERITY_NUMBER;
+    }
+    throw new IllegalStateException("Unrecognized level " + level.name());
```

Comment:
While log4j probably doesn't ever add log levels, it's conceivable and could trigger this line if log4j is newer than what we've gotten to compile against. Let's return UNDEFINED_SEVERITY_NUMBER instead of throwing